File: staffManagementUI/src/app/people/person.ts
import { PersonLinks } from './person-links';
import { PersonMeta } from './person-meta';

export class Person {

  id: string;
  userId: string;
  firstName: string;
  lastName: string;
  email: string;
  dateOfBirth?: string;
  genderId?: string;
  maritalStatusId?: string;
  numberOfDependants?: number;
  phoneNumber?: string;
  allowedSpecialShift?: boolean;
  allowedOnCall?: boolean;
  fullyTrained?: boolean;
  links: PersonLinks;
  constructor(id: string,
              userId: string,
              firstName: string,
              lastName: string,
              email: string,
              dateOfBirth: string,
              genderId: string,
              maritalStatusId: string,
              numberOfDependants: number,
              phoneNumber: string,
              allowedSpecialShift: boolean,
              allowedOnCall: boolean,
              fullyTrained: boolean,
              links: PersonLinks,
              meta: PersonMeta) {

    this.id = id;
    this.userId = userId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.email = email;
    this.dateOfBirth = dateOfBirth;
    this.genderId = genderId;
    this.maritalStatusId = maritalStatusId;
    this.numberOfDependants = numberOfDependants;
    this.phoneNumber = phoneNumber;
    this.allowedSpecialShift = allowedSpecialShift;
    this.allowedOnCall = allowedOnCall;
    this.fullyTrained = fullyTrained;
    this.links = links;
    this.meta = meta;
  }

  meta: PersonMeta;
}

File: src/main/java/com/wiltech/people/Person.java
package com.wiltech.people;

import java.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * The type Person.
 */
@Entity
@Table(name = "person")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private Long userId;

    @NotEmpty
    private String firstName;

    @NotNull
    private String lastName;

    @NotNull
    private String email;

    @Enumerated(EnumType.STRING)
    private PersonGenderType gender;

    private String phoneNumber;

    private LocalDate dateOfBirth;

    @Enumerated(EnumType.STRING)
    private PersonMaritalStatusType maritalStatus;

    private Integer numberOfDependants;

    private Boolean allowedOnCall;

    private Boolean fullyTrained;

    private Boolean allowedSpecialShift;

    /**
     * Update person.
     * @param personResource the person resource
     */
    public void updatePerson(final PersonResource personResource) {
        this.firstName = personResource.getFirstName();
        this.lastName = personResource.getLastName();
        this.email = personResource.getEmail();
        this.gender = personResource.getGenderId();
        this.phoneNumber = personResource.getPhoneNumber();
        this.dateOfBirth = personResource.getDateOfBirth();
        this.maritalStatus = personResource.getMaritalStatusId();
        this.numberOfDependants = personResource.getNumberOfDependants();
        this.allowedOnCall = personResource.getAllowedOnCall();
        this.fullyTrained = personResource.getFullyTrained();
        this.allowedSpecialShift = personResource.getAllowedSpecialShift();
    }

    public void updatePersonDetails(final String firstName, final String lastName, final String email, final LocalDate dateOfBirth) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.dateOfBirth = dateOfBirth;
    }
}
